recipes/nuraft/all/patches/0001-cmake_patches.patch

```diff
@@ -27,72 +38,55 @@ index 87685ff..1fc9199 100644
 -                    /usr/local/include
 -                    /usr/include)
 -
--    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DASIO_STANDALONE")
++    find_package(Asio CONFIG REQUIRED)
+     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DASIO_STANDALONE")
 -
 -endif ()
 -
 -if (NOT ASIO_INCLUDE_DIR)
 -    message(FATAL_ERROR "Can't find ASIO header files")
 -else ()
 -    message(STATUS "ASIO include path: " ${ASIO_INCLUDE_DIR})
--endif ()
-+#if (BOOST_INCLUDE_PATH AND BOOST_LIBRARY_PATH)
-+#    # If Boost path (both include and library) is given,
-+#    # use Boost's ASIO.
-+#    message(STATUS "Boost include path: " ${BOOST_INCLUDE_PATH})
-+#    message(STATUS "Boost library path: " ${BOOST_LIBRARY_PATH})
-+#
-+#    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DUSE_BOOST_ASIO")
-+#
-+#    set(ASIO_INCLUDE_DIR ${BOOST_INCLUDE_PATH})
-+#    set(LIBBOOST_SYSTEM "${BOOST_LIBRARY_PATH}/libboost_system.a")
-+#
-+#else ()
-+#    # If not, ASIO standalone mode.
-+#    FIND_PATH(ASIO_INCLUDE_DIR
-+#              NAME asio.hpp
-+#              HINTS ${PROJECT_SOURCE_DIR}/asio/asio/include
-+#                    $ENV{HOME}/local/include
-+#                    /opt/local/include
-+#                    /usr/local/include
-+#                    /usr/include)
-+#
-+#    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DASIO_STANDALONE")
-+#
-+#endif ()
-+
-+#if (NOT ASIO_INCLUDE_DIR)
-+#    message(FATAL_ERROR "Can't find ASIO header files")
-+#else ()
-+#    message(STATUS "ASIO include path: " ${ASIO_INCLUDE_DIR})
-+#endif ()
++    set(ASIO_DEP asio::asio)
+ endif ()
  
  
  # === Includes ===
  include_directories(BEFORE ./)
 -include_directories(BEFORE ${ASIO_INCLUDE_DIR})
-+#include_directories(BEFORE ${ASIO_INCLUDE_DIR})
  include_directories(BEFORE ${PROJECT_SOURCE_DIR}/include)
  include_directories(BEFORE ${PROJECT_SOURCE_DIR}/include/libnuraft)
  include_directories(BEFORE ${PROJECT_SOURCE_DIR}/examples)
-@@ -213,6 +231,7 @@ if (NOT (DISABLE_SSL GREATER 0))
-     )
+@@ -81,20 +64,10 @@ endif()
+ if (NOT WIN32)
+     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall")
+     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-pessimizing-move")
+-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g")
+     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
```

Comment:
This should be managed by Conan

Reply:
If I let it, then the shared library option does not include PIC and I end up with:
```
/usr/bin/ld: CMakeFiles/RAFT_CORE_OBJ.dir/src/handle_commit.cxx.o: relocation R_X86_64_TPOFF32 against `_ZZN6nuraft11raft_server11reconfigureERKSt10shared_ptrINS_14cluster_configEEE8temp_buf' can not be used when making a shared object; recompile with -fPIC
```
So it seems fPIC is needed regardless which conan is not managing correctly?

Reply:
conan does not pass -fPIC, but the `CMAKE_POSITION_INDEPENDENT_CODE` definition

https://docs.conan.io/en/latest/reference/conanfile/tools/cmake/cmaketoolchain.html#cmaketoolchain

Reply:
Correct, and it seems to not be present with the shared option enabled (I’ll double check), so I had to leave the original author’s -fPIC flag in the cmake rules.

Reply:
😕 by default CMake does the right thing https://cmake.org/cmake/help/latest/prop_tgt/POSITION_INDEPENDENT_CODE.html#prop_tgt:POSITION_INDEPENDENT_CODE

So there must be some hackery in the build script tripping up the sensible default 🙈